### PR TITLE
Fixed fixable CodeClimate issues

### PR DESCRIPTION
**Pull request by andela-toduah**

### CODE REVIEW

File: app/controllers/concerns/provider_helper.rb [hunk 1]
```diff
@@ -1,13 +1,8 @@
...
-    when /^slack/
-      '/auth/slack'
-    else
-      raise AuthProviderError.new("The provider is not supported on this platform")
-    end
-    provider_url
+    return '/auth/google_oauth2' if provider =~ /^google/
+    return '/auth/slack' if provider =~ /^slack/
+      
+    raise AuthProviderError.new("The provider is not supported on this platform")
```
Comment (andela-echigbo) [inline app/controllers/concerns/provider_helper.rb]: Nice!! 🍺

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -15,7 +15,7 @@ def show
   def update
     if @user.update(user_params)
       render :show
-    else @user.errors
+    elsif @user.errors
       render json: @user.errors, status: :unprocessable_entity
     end
```
Comment (andela-echigbo) [inline app/controllers/users_controller.rb]: Could there be a case where the update is unsuccessful and yet no error? If so, then we should have an `else` to handle that case. If not, then I think we should just change the `elsif @user.errors` to `else`.

Comment (andela-toduah) [inline app/controllers/users_controller.rb]: That's true. I've implemented the else condition.

**Comment by andela-oakinniranye** [comment]
:+1:

**Pull request merged**